src/pcre2grep.c

```diff
@@ -4421,6 +4421,8 @@ if (no_ucp) pcre2_options &= ~PCRE2_UCP;
 if (case_restrict) extra_options |= PCRE2_EXTRA_CASELESS_RESTRICT;
 if (posix_digit)
   extra_options |= (PCRE2_EXTRA_ASCII_BSD | PCRE2_EXTRA_ASCII_DIGIT);
+if ((pcre2_options & PCRE2_LITERAL) != 0)
+  extra_options &= ~PCRE2_EXTRA_NEVER_CALLOUT;
```

Comment:
Because the new flag "NEVER_CALLOUT" only affects pattern syntax, I think it's logically incompatible with PCRE2_LITERAL, because that changes the interpretation of the pattern entirely.

It would be benign to accept it and ignore it, but probably more confusing than helpful.

All the other options which are affect pattern syntax are forbidden when PCRE2_LITERAL is set.